adservertargeting/respdataprocessor.go

```diff
@@ -0,0 +1,180 @@
+		Prebid: &openrtb_ext.ExtBidPrebid{
+			Targeting: targetingDataTruncated,
+		},
+	}
+	bidExtTargeting, err := json.Marshal(bidExtTargetingData)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+		return nil
+	}
+
+	newExt, err := jsonpatch.MergePatch(bid.Ext, bidExtTargeting)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+		return nil
+	}
+	return newExt
+}
+
+func resolveKey(respTargetingData ResponseTargetingData, bidderName string) string {
+	key := respTargetingData.Key
+	if respTargetingData.HasMacro {
+		key = strings.Replace(respTargetingData.Key, bidderMacro, bidderName, -1)
+	}
+	return key
+}
+
+func truncateTargetingKeys(targetingData map[string]string, truncateTargetAttribute *int) map[string]string {
+	maxLength := MaxKeyLength
+	if truncateTargetAttribute != nil {
+		maxLength = *truncateTargetAttribute
+		if maxLength < 0 {
+			maxLength = MaxKeyLength
+		}
+	}
+
+	targetingDataTruncated := make(map[string]string)
+	for key, value := range targetingData {
+		newKey := openrtb_ext.TargetingKey(key).TruncateKey(maxLength)
+		targetingDataTruncated[newKey] = value
+	}
+	return targetingDataTruncated
+}
+
+func getValueFromSeatBidBid(
+	key, path string,
+	targetingData map[string]string,
+	bidsHolder bidsCache,
+	bidderName string,
+	bid openrtb2.Bid,
+	warnings []openrtb_ext.ExtBidderMessage) {
+	bidBytes, err := bidsHolder.GetBid(bidderName, bid.ID, bid)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+
+	}
+	bidSplit := strings.Split(path, "seatbid.bid.")
+	value, err := splitAndGet(bidSplit[1], bidBytes, pathDelimiter)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+
+	}
+	targetingData[key] = value
+}
+
+func getValueFromSeatBidExt(
+	key, path string,
+	targetingData map[string]string,
+	seatExt json.RawMessage,
+	warnings []openrtb_ext.ExtBidderMessage) {
+	seatBidSplit := strings.Split(path, "seatbid.ext.")
+	value, err := splitAndGet(seatBidSplit[1], seatExt, pathDelimiter)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+	}
+
+	targetingData[key] = value
+}
+
+func getValueFromRespExt(
+	key, path string,
+	targetingData map[string]string,
+	respExt json.RawMessage,
+	warnings []openrtb_ext.ExtBidderMessage) {
+	//path points to resp.ext, means path starts with ext
+	extSplit := strings.Split(path, "ext.")
+	value, err := splitAndGet(extSplit[1], respExt, pathDelimiter)
+	if err != nil {
+		warnings = append(warnings, createWarning(err.Error()))
+	}
+	targetingData[key] = value
+}
+
+func getValueFromResp(
+	key, path string,
+	targetingData map[string]string,
+	bidderRespHolder respCache,
+	response *openrtb2.BidResponse,
+	warnings []openrtb_ext.ExtBidderMessage) {
+	// path points to value in response, not seat or ext
+	// bidder response has limited number of properties
+	// instead of unmarshal the whole response with seats and bids
+	// try to find data by field name
```

Comment:
Agree, moved

Reply:
Not really what I was thinking, but it could work too. I meant to suggest something like this:

```
// getValueFromResp optimizes retrieval of paths for a bid response (not a seat or ext)
func getValueFromResp(
```

Reply:
Oh ok, modified